packages/lib-classifier/src/components/Classifier/components/Feedback/components/Graph2dRangeFeedback.js

```diff
@@ -0,0 +1,116 @@
+  const { 
+    applicableRules,
+    showModal: feedback
+  } = stores.classifierStore.feedback
+  const {
+    active: subject
+  } = stores.classifierStore.subjects
+
+  return {
+    annotations,
+    applicableRules,
+    feedback,
+    subject
+  }
+}
+
+@inject(storeMapper)
+@observer
+class Graph2dRangeFeedback extends Component {
+  constructor () {
+    super()
+
+    this.drawFeedbackBrushes = this.drawFeedbackBrushes.bind(this)
+  }
+
+  drawFeedbackBrushes (d3annotationsLayer, repositionBrush) {
+    const annotationBrushes = []
+    this.props.annotations.forEach(annotation => {
+      const { task, value } = toJS(annotation)
+      value.forEach((marking, i) => {
+        const markingBrush = {
+          id: i,
+          brush: d3.brushX(),
+          maxX: (marking.x + (marking.width / 2)),
+          minX: (marking.x - (marking.width / 2))
+        }
+        annotationBrushes.push(markingBrush)
+      })
+    })
+
+    const ruleBrushes = []
+    this.props.applicableRules.forEach(rule => {
+      const ruleBrush = {
+        id: rule.id,
+        brush: d3.brushX(),
+        maxX: (parseInt(rule.x, 10) + (parseInt(rule.width, 10) / 2) + parseInt(rule.tolerance, 10)),
+        minX: (parseInt(rule.x, 10) - (parseInt(rule.width, 10) / 2) - parseInt(rule.tolerance, 10)),
+        success: rule.success
+      }
+      ruleBrushes.push(ruleBrush)
+    })
+
+    const feedbackBrushes = annotationBrushes.concat(ruleBrushes)
+
+    // Join the D3 brush objects with our internal annotationBrushes array
+    const brushSelection = d3annotationsLayer
+      .selectAll('.brush')
+      .data(feedbackBrushes, (d) => d.id)
+
+    // Set up new brushes
+    brushSelection.enter()
+      .insert('g', '.brush')
+      .attr('class', 'brush')
+      .attr('id', (brush) => (`brush-${brush.id}`))
+      .each(function applyBrushLogic (feedbackBrush) { // Don't use ()=>{}
+        feedbackBrush.brush(d3.select(this)) // Apply the brush logic to the <g.brush> element (i.e. 'this')
+      })
+
+    // Modify brush fill color...
+    brushSelection
+      .each(function fill (feedbackBrush) { // Don't use ()=>{}
+        d3.select(this)
+          .attr('class', 'brush')
+          .selectAll('.selection')
+          .style('fill', () => {
+            if (feedbackBrush.success === true) {
+              return 'green'
+            } else if (feedbackBrush.success === false) {
+              return 'red'
+            } else {
+              return 'white'
+            }
+          })
+      })
+
+    // Reposition/re-draw brushes
+    feedbackBrushes.forEach((feedbackBrush) => {
+      const d3brush = d3annotationsLayer.select(`#brush-${feedbackBrush.id}`)
+
+      repositionBrush(feedbackBrush, d3brush)
+    })
+  }
+
+  render () {
+    const { feedback, subject } = this.props
+    return (
+      <LightCurveViewer 
+        drawFeedbackBrushes={this.drawFeedbackBrushes}
+        feedback={feedback}
```

Comment:
I don't like adding `drawFeedbackBrushes` or `feedback` as props, considered adding to the feedback store, but the thought of a workflow with different types of feedback strategies (i.e. "single answer question" and "radial") haunts me, which would complicate what would be added to the feedback store so different types of feedback markings could be shown in the subject viewer.

Reply:
Looking at the feedback code in PFE, I think what you’re looking for here is a Strategy pattern. 

So, the viewer loops through annotations and invokes an abstract renderer that should render a mark for that annotation, based on the annotation type. Each feedback strategy then defines a method that renders a feedback mark for an annotation.

Reply:
I think [this code in PFE](https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/classifier/tasks/drawing/markings-renderer.cjsx#L49-L53) is missing from the current LCV, or maybe the DataVisAnnotationTask, but I'm doing something similar albeit not ideally with `drawFeedbackBrushes`. 

I feel like the drawing tasks and the subject viewer are decoupled differently in PFE than the `dataVisAnnotationTask` and the LCV, but maybe I'm wrong or it shouldn't matter.

Reply:
That's exactly the code I was thinking of. The same pattern should apply here, I think.

Reply:
So I think I'd need to refactor the LCV, I don't think it's currently configured to create markings from existing annotations? per https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewer.js#L214

Reply:
`annotations` are passed in as props, which is all you need to get the strategies right?

Reply:
I don't think the annotations coming into the LCV from the classifier store include feedback information, I think in PFE we [edit the annotations so they do include feedback](https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/features/feedback/classifier/helpers/open-feedback-modal.jsx#L51-L55), then pass those edited annotations (with feedback) down through the tree as props.

Reply:
Oh, that's a pain. I was thinking the code would be something like
```javascript
annotations.forEach(function (annotation) {
  strategies[annotation.strategy].drawFeedback()
}
```
but I hadn't thought about how rules map to annotations.